pkgs/test/stdenv/default.nix

```diff
@@ -142,6 +142,13 @@ in
     '';
   };
 
+  # Check that mkDerivation rejects MD5 hashes
+  rejectedHashes = lib.recurseIntoAttrs {
+    md5 = testers.testBuildFailure (runCommand "md5 outputHash rejected" {
+      outputHash = "md5-fPt7dxVVP7ffY3MxkQdwVw==";
+    } "true");
+  };
+
```

Comment:
This doesn't really work as you expect. `testBuildFailure` can only "catch" build failures, but the expression here already fails to evaluate. I think the previous test with `tryEval` should be done instead.